libraries/chain/transaction.cpp

```diff
@@ -92,7 +92,7 @@ fc::microseconds transaction::get_signature_keys( const vector<signature_type>&
                   // If public key can be recovered from signature
                   auto [itr, ok] = recovered_pub_keys.emplace(sig, digest);
                   SYS_ASSERT( allow_duplicate_keys || ok, tx_duplicate_sig, "duplicate signature for key ${k}", ("k", *itr) );
-               } else {
+               } else if constexpr ( !std::is_same_v<Shim, fc::crypto::bls::signature_shim>) {
```

Comment:
I think we should `SYS_ASSERT` for a `bls::signature_shim` has this silently does nothing and returns an empty `recovered_pub_keys`. It is not clear to me without a test that this would then fail when the transaction authority was checked. If the transaction didn't specify an authority then the transaction might be allowed.

If we plan to support BLS `fc::crypto::public_key` then seems we should support recover key. Likely need some kind of transaction extension like `ed`. Along with full test coverage.

If we are not going to support `fc::crypto::public_key` for transactions then I question adding it as one of the supported `fc::crypto::public_key` types.